components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -149,6 +155,18 @@ public void handle(List<String> roles, String subject, Map<String, String> attri
                  */
                 IdentityUtil.threadLocalProperties.get().put(FrameworkConstants.JIT_PROVISIONING_FLOW, true);
                 if (!userClaims.isEmpty()) {
+
+                    List<Claim> existingUserClaimList = new ArrayList<>(Arrays.asList(userStoreManager
+                            .getUserClaimValues(UserCoreUtil.removeDomainFromName(username), null)));
```

Comment:
Better to separate out the logic in here. This reduces the code readability. WDYT?

Reply:
Addressed with commit a899a6042f14ec3493540d2dbfe141e3fd0ec503